### PR TITLE
Enhance Image and ActionableButton Widget to add ability to copy text

### LINKED ISSUE #3033: [Feature] Add copy text widget
**Describe the feature request.**
Add a copy text widget that can be configurable in the user profile configs 

**Additional context**
In the referral use case for ThinkMD integration, we should add the ability to be able to copy the referral link URL once the user accepts the concept form to perform a ThinkMD assessment 

**Acceptance criteria**
The user should be able to copy text appended to the user profile  once the user clicks on the copy text widget  

**Implementation plan (For Engineers)**
The plan for implementing the solution e.g. via a _description_ or a _checklist_ for the various ordered tasks that must be completed.
_i.e. Describe how you intend to solve the problem_

### PR DESCRIPTION
**Pull request by SebaMutuku**
**IMPORTANT: Where possible all PRs must be linked to a Github issue**

Fixes #3033 

**Engineer Checklist**
- [x] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [x] I have added any strings visible on UI components to the `strings.xml` file
- [ ] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [ ] I have built and run the FHIRCore app to verify my change fixes the issue and/or does not break the app 
- [x] I have checked that this PR does NOT introduce **breaking changes** that require an update to **_Content_** and/or **_Configs_**? _If it does add a sample here or a link to exactly what changes need to be made to the content._

**Code Reviewer Checklist**
- [ ] I have verified **Unit tests** have been written for any new feature(s) and edge cases
- [ ] I have verified any strings visible on UI components are in the `strings.xml` file
- [ ] I have verifed the [CHANGELOG.md](./CHANGELOG.md) file has any notable changes to the codebase
- [ ] I have verified the solution has been implemented in a configurable and generic way for reuseable components
- [ ] I have built and run the FHIRCore app to verify the change fixes the issue and/or does not break the app
 

Sample config for this 

``` json 
   {
          "viewType": "CARD",
          "padding": 16,
          "visible": "true",
          "header": {
            "viewType": "COMPOUND_TEXT",
            "primaryText": "THINKMD TASKS",
            "primaryTextColor": "#6F7274",
            "fontSize": 18.0,
            "padding": 16
          },
          "content": [
            {
              "viewType": "BUTTON",
              "smallSized": "true",
              "text": "Share ThinkMD Self Assessment link",
              "status": "DUE",
              "visible": "true",
              "enabled": "@{patientActive}",
              "startIcon": {
                "type": "local",
                "reference": "ic_copy",
                "color": "#0000FF"
              },
              "actions": [
                {
                  "trigger": "ON_CLICK",
                  "workflow": "COPY_TEXT",
                  "params": [
                    {
                      "paramType": "PARAMDATA",
                      "key": "copyText",
                      "value": "https://jakarta-thinkmd-v4-app-2hqzu73qzq-et.a.run.app/#/start/@{patientId}"
                    }
                  ]
                }
              ]
            }
          ]
        },
```

### CODE REVIEW

File: android/quest/src/androidTest/java/org/smartregister/fhircore/quest/integration/ui/shared/components/ActionableButtonTest.kt [hunk 1]
```diff
@@ -100,6 +100,17 @@ class ActionableButtonTest {
...
 
+  @Test
+  fun testActionableButtonRendersAndCopyText() {
+    setCopyButtonContent("false")
+
+    composeRule
+      .onNodeWithText("Copy Button Text", useUnmergedTree = true)
+      .assertExists()
+      .assertIsDisplayed()
+      .performClick()
```
Comment (ellykits) [inline android/quest/src/androidTest/java/org/smartregister/fhircore/quest/integration/ui/shared/components/ActionableButtonTest.kt:111]: Should there be a verification done for this perform click action?

**Review by ellykits — COMMENTED** [review]
:+1: LGTM. Just a simple comment on the test case.

File: android/quest/src/androidTest/java/org/smartregister/fhircore/quest/integration/ui/shared/components/ActionableButtonTest.kt (hunk 1 above)
Comment (SebaMutuku) [reply android/quest/src/androidTest/java/org/smartregister/fhircore/quest/integration/ui/shared/components/ActionableButtonTest.kt:111]: I will update the test to test on click function instead

**Comment by SebaMutuku** [comment]
@MargaretNjenga @f-odhiambo , for the icon text, we might have to go with a text instead. Adding the link there might jump it to 3 linkes.
Here's a sample copy link option 
image

Then a toast shows when one clicks on the icon, below message appears
image

Kindly advise if there's anything needed to change as I fix failing tests

**Comment by f-odhiambo** [comment]
@SebaMutuku Can we add the 'Tap icon ...' text as normal text and not make it look like a hyperlink? Rest of the stuff LGTM

**Comment by SebaMutuku** [comment]
@f-odhiambo sure. Updating the same

**Comment by f-odhiambo** [comment]
> @f-odhiambo This has been updated image

Let make the other text grey

**Comment by SebaMutuku** [comment]
@f-odhiambo this is configurable. I only used that color for visibility

**Comment by pld** [comment]
@f-odhiambo @SebaMutuku I have marked this as blocked, it needs design review, fine to continue on functional requirements but it is not efficient to work on the design until @rowo and @HenryRae have reviewed, thank you

**Comment by HenryRae** [comment]
@SebaMutuku @MargaretNjenga what are we trying to archive here with the  copy link on the profile?

**Comment by HenryRae** [comment]
Here's a [link](https://www.figma.com/file/HWjVl0CiRFc5hYV4SD0Bfu/SID-Bunda-App---FHIR-Core-Module-Designs?type=design&node-id=1584%3A27037&mode=design&t=i9409jRjgJSvBQyo-1) to the proposed design.

image
cc: @SebaMutuku @MargaretNjenga @f-odhiambo

**Comment by SebaMutuku** [comment]
@HenryRae this has been effected.
cc @f-odhiambo @pld @MargaretNjenga 

image

**Review by pld — COMMENTED** [review]
looks good! @SebaMutuku can you update the docs to show how to add this copy text link to an app?

**Comment by SebaMutuku** [comment]
@pld updated.

**Review by pld — DISMISSED** [review]

**Review by ellykits — DISMISSED** [review]
LGTM :+1:

**Comment by ellykits** [comment]
Address the CI issues @SebaMutuku so we can merge this.

**Review by ellykits — APPROVED** [review]

**Pull request merged**